lib/compat/wordpress-6.2/class-gutenberg-rest-navigation-controller.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Templates_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Templates REST API Controller.
+ */
+class Gutenberg_REST_Navigation_Controller extends WP_REST_Posts_Controller {
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+
+		parent::register_routes();
+
+		// Lists a single nav item based on the given id or slug.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/((?P<id>[\d]+)|(?P<slug>[\w\-]+))',
```

Comment:
Both id and slug are needed in this request otherwise backwards compatibility is broken.